File: solutions/csharp/bob/2/Bob.cs
using System;
using System.Linq;

public static class Bob
{
    public static string Response(string statement)
    {
        statement = statement.Trim();

        switch (statement.EndsWith("?"), LettersAreUpper(statement), statement == "")
        {
            case (true, false, _):
                return "Sure.";
            case (false, true, _):
                return "Whoa, chill out!";
            case (true, true, _):
                return "Calm down, I know what I'm doing!";
            case (_, _, true):
                return "Fine. Be that way!";
            default:
                return "Whatever.";
        }
    }

    private static bool LettersAreUpper(string text) =>
        text.Where(c => Char.IsLetter(c))
            .DefaultIfEmpty('c') // to counter that .All return true for empty collection.
            .All(c => Char.IsUpper(c));
}

File: solutions/python/bob/2/bob.py
def response(hey_bob: str) -> str:
    hey_bob = hey_bob.strip()
    match (hey_bob.endswith("?"), hey_bob.isupper(), hey_bob == ""):
        case (True, False, _):
            return "Sure."
        case (False, True, _):
            return "Whoa, chill out!"
        case (True, True, _):
            return "Calm down, I know what I'm doing!"
        case (_, _, True):
            return "Fine. Be that way!"
        case _:
            return "Whatever."
